README.md

```diff
@@ -139,6 +139,8 @@ For example
 < ...
 ```
 
+Note that `ls` support is OPTIONAL, i.e. implementations MAY support sending and/or receiving `ls`. This in turn implies that implementations MUST NOT depend on a remote node supporting `ls`.
```

Comment:
Should this be at the beginning of this section?

> Implementations MAY support "listing" of the available protocols. A list message is simply:

Reply:
(I'd also consider marking it as deprecated, but we can punt that till later)

Reply:
> Should this be at the beginning of this section?
> 
> > Implementations MAY support "listing" of the available protocols. A list message is simply:

Sounds good. Added via https://github.com/multiformats/multistream-select/pull/22/commits/94f37eaabd1f006ad7073a226ce23d6ff6327454.

> (I'd also consider marking it as deprecated, but we can punt that till later)

I am fine with deprecating `ls`. I don't feel strongly enough about it to push that effort forward though.